R/TealAppDriver.R

```diff
@@ -45,25 +45,51 @@ TealAppDriver <- R6::R6Class( # nolint: object_name.
     initialize = function(data,
                           modules,
                           filter = teal_slices(),
-                          title = build_app_title(),
+                          title_args = list(),
                           header = tags$p(),
                           footer = tags$p(),
-                          landing_popup = NULL,
+                          landing_popup_args = NULL,
                           timeout = rlang::missing_arg(),
                           load_timeout = rlang::missing_arg(),
                           ...) {
       private$data <- data
       private$modules <- modules
       private$filter <- filter
+
+      new_title <- modifyList(
+        list(
+          title = "Custom Teal App Title",
+          favicon = "https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/PNG/teal.png"
+        ),
+        title_args
+      )
       app <- init(
         data = data,
         modules = modules,
-        filter = filter,
-        title = title,
-        header = header,
-        footer = footer,
-        landing_popup = landing_popup,
-      )
+        filter = filter
+      ) |>
+        modify_title(title = new_title$title, favicon = new_title$favicon) |>
+        modify_header(header) |>
+        modify_footer(footer)
+
+      if (!is.null(landing_popup_args)) {
+        args <- modifyList(
+          list(
+            id = "landingpopup",
+            title = NULL,
+            content = NULL,
+            buttons = modalButton("Accept")
+          ),
+          landing_popup_args
+        )
+        app <- app |>
+          add_landing_popup(
```

Comment:
To introduce the pipe operator in the internal code we need to agree to certain style rules within a team. Pipe might be overused to replace a single call which in my opinion is bad as it doesn't bring clarity plus it is a extra sugar to evaluate.

```suggestion
        app <- add_landing_popup(
          app,
```